djangocms_bootstrap4/contrib/bootstrap4_grid/models.py

```diff
@@ -0,0 +1,249 @@
+    )
+    tag_type = TagTypeField()
+    attributes = AttributesField()
+
+    def __str__(self):
+        return str(self.pk)
+
+    def get_short_description(self):
+        instance = self.get_plugin_instance()[0]
+
+        if not instance:
+            return ugettext('<empty>')
+
+        column_count = len(self.child_plugin_instances or [])
+        column_count_str = ungettext(
+            '(1 column)',
+            '(%(count)i columns)',
+            column_count
+        ) % {'count': column_count}
+        column_count_str += ' .{}'.format(
+            ' .'.join(instance.attributes['class'].split())
+        )
+
+        return column_count_str
+
+
+@python_2_unicode_compatible
+class Bootstrap4GridColumn(CMSPlugin):
+    """
+    Layout > Grid: "Column" Plugin
+    https://getbootstrap.com/docs/4.0/layout/grid/
+    """
+    column_type = models.CharField(
+        verbose_name=_('Column type'),
+        choices=GRID_COLUMN_CHOICES,
+        default=GRID_COLUMN_CHOICES[0][0],
+        blank=True,
+        max_length=255,
+    )
+    column_size = IntegerRangeField(
+        verbose_name=_('Columne size'),
+        blank=True,
+        null=True,
+        min_value=0,
+        max_value=GRID_SIZE,
+        help_text=_(
+            'Nummeric value from 1 - {bound}. '
+            'Spreads the columns evenly when empty.').format(bound=GRID_SIZE)
+    )
+    column_alignment = models.CharField(
+        verbose_name=_('Alignment'),
+        choices=GRID_COLUMN_ALIGNMENT,
+        blank=True,
+        max_length=255,
+    )
+    tag_type = TagTypeField()
+    attributes = AttributesField()
+
+    def __str__(self):
+        return str(self.pk)
+
+    def get_short_description(self):
+        text = ''
+        classes = self.get_grid_values();
+        if self.column_size:
+            text += '(col-{}) '.format(self.column_size)
+        else:
+            text += '(auto) '
+        if self.column_type != 'col':
+            text += '.{} '.format(self.column_type)
+        if classes:
+            text += '.{}'.format(' .'.join(self.get_grid_values()))
+        return text
+
+    def get_grid_values(self):
+        classes = []
+        for device in DEVICE_SIZES:
+            for element in ('col', 'order', 'ml', 'mr'):
+                size = getattr(self, '{}_{}'.format(device, element))
+                if size and (element == 'col' or element == 'order'):
+                    classes.append('{}-{}-{}'.format(element, device, int(size)))
+                elif size:
+                    classes.append('{}-{}-{}'.format(element, device, 'auto'))
+        return classes
+
+
+IntegerRangeFieldPartial = partial(
+    IntegerRangeField,
+    blank=True,
+    null=True,
+    max_value=GRID_SIZE,
+)
+
+BooleanFieldPartial = partial(
+    models.BooleanField,
+    default=False,
+)
+
+# Loop through Bootstrap 4 device choices and generate
+# model fields to cover col-*, order-*
+for size in DEVICE_SIZES:
```

Comment:
this won't really work.
fields require migrations and because device sizes is configurable, if a developer changes this setting in their project, django will require a migration but the migration will have to live on this app.

If device sizes has x,y,z as options, then I suggest creating the columns for all those but on the UI form, limit it to the configured choices.

Reply:
I can live with that, if someone changes it to have their own migrations. But let's eventually chat on how to solve it as you suggest

Reply:
Adapted constants